admin/management/views.py

```diff
@@ -120,7 +120,14 @@ def post(self, request, *args, **kwargs):
         if monthly_report_date:
             report_date = isoparse(monthly_report_date).date()
         else:
-            report_date = datetime.datetime.now().date()
+            messages.error(request, 'Monthly Reporter can\'t be run in future or current month')
+            return redirect(reverse('management:commands'))
+
+        today = datetime.date.today()
+
+        if report_date >= today.replace(day=1):
+            messages.error(request, 'Monthly Reporter can\'t be run for the current or a future month.')
+            return redirect(reverse('management:commands'))
```

Comment:
This is a good idea to prevent bad input, but with above changes would just need `if report_date and <rest of conditional>`